From cc892a19dfe49ee10d48ada61641de46db1d1100 Mon Sep 17 00:00:00 2001
From: Martin Jansa <Martin.Jansa@gmail.com>
Date: Wed, 1 Feb 2023 18:42:49 +0100
Subject: [PATCH] ppapi: Fix build with gcc-13

Fixes:
FAILED: obj/ppapi/proxy/proxy/ppb_graphics_3d_proxy.o
x86_64-webos-linux-g++  -m64 -march=nehalem -mtune=generic -mfpmath=sse -msse4.2 --sysroot=/OE/build/luneos-mickledore/webos-ports/tmp-glibc/work/corei7-64-webos-linux/webruntime/94-r59.2/recipe-sysroot -MMD -MF obj/ppapi/proxy/proxy/ppb_
graphics_3d_proxy.o.d -DENABLE_BROWSER_CONTROL_WEBAPI=1 -DENABLE_SAMPLE_WEBAPI=1 -DENABLE_MEMORYMANAGER_WEBAPI=1 -DENABLE_WEBOS_SERVICE_BRIDGE_WEBAPI=1 -DENABLE_WEBOS_SYSTEM_WEBAPI=1 -DNEVA_DCHECK_ALWAYS_ON=1 -DOZONE_PLATFORM_WAYLAND_EXTE
RNAL=1 -DNEVA_OZONE_PLATFORM_WAYLAND=1 -DOS_WEBOS=1 -DNEVA_VIDEO_HOLE=1 -DUSE_NEVA_APPRUNTIME=1 -DUSE_NEVA_BROWSER_SERVICE=1 -DUSE_WEBRISK_SERVICE=1 -DENABLE_WEBM_VIDEO_CODECS=1 -DUSE_CBE=1 -DUSE_PMLOG=1 -DUSE_NEVA_SUSPEND_MEDIA_CAPTURE=1
 -DUSE_FILESCHEME_CODECACHE=1 -DWEBOS_SUBMISSION_NUMBER=13 -DUSE_LOCAL_STORAGE_TRACKER=1 -DUSE_UDEV -DUSE_AURA=1 -DUSE_GLIB=1 -DUSE_NSS_CERTS=1 -DUSE_OZONE=1 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D__STDC_CONSTA
NT_MACROS -D__STDC_FORMAT_MACROS -D_FORTIFY_SOURCE=2 -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -DPPAPI_PROXY_IMPLEMENTATION -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_40 -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_40 -DWEBP_EXTER
N=extern -DUSE_EGL -DU_USING_ICU_NAMESPACE=0 -DU_ENABLE_DYLOAD=0 -DUSE_CHROMIUM_ICU=1 -DU_ENABLE_TRACING=1 -DU_ENABLE_RESOURCE_TRACING=0 -DU_STATIC_IMPLEMENTATION -DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_FILE -DGOOGLE_PROTOBUF_NO_RTTI -DGOOGLE_
PROTOBUF_NO_STATIC_INITIALIZER -DHAVE_PTHREAD -DSK_UNTIL_CRBUG_1187654_IS_FIXED -DSK_CODEC_DECODES_PNG -DSK_CODEC_DECODES_WEBP -DSK_ENCODE_PNG -DSK_ENCODE_WEBP -DSK_ENABLE_SKSL -DSK_USER_CONFIG_HEADER=\"../../skia/config/SkUserConfig.h\"
-DSK_GL -DSK_CODEC_DECODES_JPEG -DSK_ENCODE_JPEG -DSK_HAS_WUFFS_LIBRARY -DSK_VULKAN_HEADER=\"../../skia/config/SkVulkanConfig.h\" -DSK_VULKAN=1 -DSK_SUPPORT_GPU=1 -DSK_GPU_WORKAROUNDS_HEADER=\"gpu/config/gpu_driver_bug_workaround_autogen.
h\" -DV8_COMPRESS_POINTERS -DV8_COMPRESS_POINTERS_IN_SHARED_CAGE -DV8_31BIT_SMIS_ON_64BIT_ARCH -DV8_DEPRECATION_WARNINGS -DCPPGC_CAGED_HEAP -I../../git/src/neva -Igen/neva -I../../git/src -Igen -I../../git/src/third_party/perfetto/include
 -Igen/third_party/perfetto/build_config -Igen/third_party/perfetto -I../../git/src/third_party/libwebp/src -I../../git/src/third_party/khronos -I../../git/src/gpu -I../../git/src/third_party/abseil-cpp -I../../git/src/third_party/borings
sl/src/include -I../../git/src/third_party/protobuf/src -Igen/protoc_out -I../../git/src/third_party/ced/src -I../../git/src/third_party/icu/source/common -I../../git/src/third_party/icu/source/i18n -I../../git/src/third_party/skia -I../.
./git/src/third_party/wuffs/src/release/c -I../../git/src/third_party/vulkan/include -I../../git/src/third_party/vulkan-deps/vulkan-headers/src/include -I../../git/src/v8/include -Igen/v8/include -Wno-unused-variable -fno-ident -fno-stric
t-aliasing --param=ssp-buffer-size=4 -fstack-protector -funwind-tables -fPIC -pipe -pthread -m64 -march=x86-64 -msse3 -Wno-builtin-macro-redefined -D__DATE__= -D__TIME__= -D__TIMESTAMP__= -Wall -Wno-unused-local-typedefs -Wno-maybe-uninit
ialized -Wno-deprecated-declarations -Wno-comments -Wno-packed-not-aligned -Wno-attributes -Wno-missing-field-initializers -Wno-unused-parameter -O2 -fdata-sections -ffunction-sections -fno-omit-frame-pointer -g0 -fvisibility=hidden -I/OE
/build/luneos-mickledore/webos-ports/tmp-glibc/work/corei7-64-webos-linux/webruntime/94-r59.2/recipe-sysroot/usr/include/glib-2.0 -I/OE/build/luneos-mickledore/webos-ports/tmp-glibc/work/corei7-64-webos-linux/webruntime/94-r59.2/recipe-sy
sroot/usr/lib/glib-2.0/include -std=gnu++14 -Wno-narrowing -Wno-class-memaccess -fno-exceptions -fno-rtti -fvisibility-inlines-hidden -c ../../git/src/ppapi/proxy/ppb_graphics_3d_proxy.cc -o obj/ppapi/proxy/proxy/ppb_graphics_3d_proxy.o
In file included from ../../git/src/ppapi/utility/completion_callback_factory.h:11,
                 from ../../git/src/ppapi/proxy/proxy_completion_callback_factory.h:13,
                 from ../../git/src/ppapi/proxy/ppb_graphics_3d_proxy.h:19,
                 from ../../git/src/ppapi/proxy/ppb_graphics_3d_proxy.cc:5:
../../git/src/ppapi/utility/completion_callback_factory_thread_traits.h:70:20: error: declaration of 'typedef class pp::Lock pp::ThreadSafeThreadTraits::Lock' changes meaning of 'Lock' [-fpermissive]
   70 |   typedef pp::Lock Lock;
      |                    ^~~~
../../git/src/ppapi/utility/completion_callback_factory_thread_traits.h:66:5: note: used here to mean 'class pp::Lock'
   66 |     Lock lock_;
      |     ^~~~
In file included from ../../git/src/ppapi/utility/completion_callback_factory_thread_traits.h:12:
../../git/src/ppapi/utility/threading/lock.h:20:7: note: declared here
   20 | class Lock {
      |       ^~~~

Change-Id: I0dfa1f325fc42eb2ab10e6b6acc203d3eb5be63b
Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
Upstream-Status: Submitted [http://gpro.lge.com/c/webosose/chromium94/+/344632 ppapi: Fix build with gcc-13]

 src/ppapi/utility/completion_callback_factory_thread_traits.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ppapi/utility/completion_callback_factory_thread_traits.h b/src/ppapi/utility/completion_callback_factory_thread_traits.h
index b4fdb9a6d3..458bc4a05b 100644
--- a/src/ppapi/utility/completion_callback_factory_thread_traits.h
+++ b/src/ppapi/utility/completion_callback_factory_thread_traits.h
@@ -63,7 +63,7 @@ class ThreadSafeThreadTraits {
     }
 
    private:
-    Lock lock_;
+    pp::Lock lock_;
     int32_t ref_;
   };
 
